### PR TITLE
ENH allowed to have multiple simulated test configs to test solvers

### PR DESCRIPTION
**Pull request by zaccharieramzi**
This is a direct follow-up to https://github.com/benchopt/benchopt/pull/337 (it's not dependent as it could be merged without the latter one being merged)

I am setting this to draft since I did not craft the appropriate documentation for this new possibility.

### CODE REVIEW

**Comment by zaccharieramzi** [comment]
I don't know why the pypi test is failing.

The logs:
```
Run fountainhead/action-wait-for-check@v1.0.0
  with:
    token: ***
    checkName: report_test
    timeoutSeconds: 1600
    intervalSeconds: 30
    ref: 9117fe9a3dce38b5949b[2](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:2)a820b819451a64bcf[3](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:3)b
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.12/x6[4](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:4)
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b[5](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:5)949b2a820b819451a[6](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:6)4bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@911[7](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:7)fe9a3dce3[8](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:8)b5[9](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:9)49b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9[11](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:11)7fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@91[17](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:17)fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b8[19](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:19)451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a8[20](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:20)b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for [30](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:30) seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce[38](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:38)b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819[45](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:45)1a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b59[49](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:49)b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b8194[51](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:51)a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b[59](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:59)49b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a[64](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:64)bcf3b...
Retrieved 0 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b819451a64bcf3b...
Retrieved 1 check runs named report_test
No completed checks named report_test, waiting for 30 seconds...
Retrieving check runs named report_test on benchopt/benchopt@9117fe9a3dce38b5949b2a820b[81](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:81)[94](https://github.com/benchopt/benchopt/runs/6197766842?check_suite_focus=true#step:7:94)51a64bcf3b...
```

Something about retrieving checks for `report_test` ...

**Comment by tomMoral** [comment]
This was a glitch, I relaunched the failing test and this worked.

File: benchopt/cli/main.py [hunk 1]
```diff
@@ -301,7 +301,9 @@ def install(benchmark, minimal, solver_names, dataset_names, force=False,
 
 
 @main.command(
-    help="Test a benchmark for benchopt.",
+    help="Test a benchmark for benchopt. It must feature a simulated dataset."
+    "For more info about the simulated dataset configurations, see"
+    "benchopt.github.io/how.html#example-of-parametrized-simulated-dataset",
```
Comment (tomMoral) [inline benchopt/cli/main.py:306]:
```suggestion
    help="Test a benchmark for benchopt. The benchmark must feature a simulated "
    "dataset. For more info about the simulated dataset configurations, see"
    "benchopt.github.io/how.html#example-of-parametrized-simulated-dataset",
```

**Review by tomMoral — COMMENTED** [review]
Maybe I am missing something but not sure why we need this new `MANY_SIMULATED_BENCHMARK`?

**Review by tomMoral — APPROVED** [review]
LGTM, thx @zaccharieramzi .

**Comment by zaccharieramzi** [comment]
@tomMoral I think in the current merged version we do not have a test on the multiple configurations in either test benchmark.
Should we create a follow up PR and use that or leave it as is for now?

**Pull request merged**